api/src/controllers/africas-talking.js

```diff
@@ -26,73 +26,59 @@ const STATUS_CODES = {
   Failed: 'failed'
 };
 
-// Stolen from: https://github.com/expressjs/express/issues/2518#issuecomment-455775771
-const isLocal = req => req.connection.localAddress === req.connection.remoteAddress;
-
-const validateSource = req => {
-  // allow "http" requests only from localhost
-  if (isLocal(req)) {
-    return;
-  }
-  if (req.protocol !== 'https') {
-    // this should be blocked by our stack but adding here to double
-    // check and prevent IP spoofing
-    logger.warn(`Attempt to access Africa's Talking endpoint with protocol: "${req.protocol}", ${req.ip}`);
-    return { code: 403, message: 'https is required' };
-  }
-  const settings = config.get('sms');
-  const ips = settings &&
-              settings.africas_talking &&
-              settings.africas_talking.allowed_ips;
-  if (!ips || !Array.isArray(ips) || !ips.includes(req.ip)) {
-    logger.warn(`Attempt to access Africa's Talking endpoint by IP address not on allowed list: "${req.ip}"`);
-    return { code: 403, message: 'Unknown IP address' };
-  }
+const getIncomingKey = () => {
+  return secureSettings.getCredentials('africastalking.com:incoming').then(key => {
```

Comment:
There's a bug in `@medic/settings` where `getCredentials` throws an error if the returning config is an empty string: 
```
ERROR: Server error: { TypeError: Cannot read property '1' of null
    at parseResponse (/home/diana/projects/medic/shared-libs/settings/src/index.js:7:69)
    at processTicksAndRejections (internal/process/task_queues.js:86:5)
  [stack]:
   "TypeError: Cannot read property '1' of null\n    at parseResponse (/home/diana/projects/medic/shared-libs/settings/src/index.js:7:69)\n    at processTicksAndRejections (internal/process/task_queues.js:86:5)",
  [message]: "Cannot read property '1' of null" } 
```
This causes the response to a request to be `"Server error"` with no additional information. 
Maybe it's a quick enough fix to include here?

Reply:
Nice catch. I'll raise an issue for that separately. I don't want to take any chances on this patch.